source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -5,12 +5,293 @@
+  by blocks of defined :term:`data layout`.
+
+Table provide read-only access to the data inside it for thread-safety reasons
+and better integration with external entities, thus, table concept is
+:term:`immutable <Immutability>`.
+
+This concept may also have different logical organization and physical
+:term:`format of the data <data format>` inside it.
+
+- Logically, table is a :ref:`dataset` with :math:`n` rows and
+  :math:`p` columns. Each row represent an :term:`observation` and each column
+  is a :term:`feature` of a dataset. Thus, table contains :math:`(n \times p)`
+  data elements, while the original (physical) amount of bytes needed to store
+  the data can be different.
+
+- Physically, table can be organized in different ways: as a :term:`homogeneous
+  <Homogeneous data>`, :term:`contiguous <Contiguous data>` array of bytes, as
+  :term:`heterogeneous <Heterogeneous data>` list of arrays of different
+  :term:`data types <Data type>`, in a compressed-sparse-row format, etc.
+
+For details, see :ref:`tables` section.
+
+.. _metadata:
+
+Metadata
+--------
+
+Metadata concept is assotiated with a :ref:`dataset` and holds an information
+about its structure and type. This information shall be enough to determine the
+particular type of a dataset, and helps to understand how to interact with
+dataset in |dal_short_name| (for example, how to use it on a particular stage of
+:ref:`data analytics pipeline <data_analytics_pipeline>` or how to access its
+data).
+
+Metadata for all datasets shall contain:
+
+- The number of rows :math:`n` and columns :math:`p` in a dataset
+
+- Type of each :term:`feature` (e.g. :term:`nominal <Nominal feature>`,
+  :term:`interval <Interval feature>`)
+
+- :term:`data type` of each feature
+
+- The kind of a dataset (e.g. :ref:`table` or :ref:`data-source`)
+
+.. note::
+  Metadata can contain both compile-time and run-time information. For example,
+  basic compile-time metadata is the type of a dataset - whether it is a
+  particular :ref:`data-source` or a :ref:`table`. Run-time information can
+  contain the :term:`feature` types and :term:`data types <Data type>` of a
+  dataset.
+
+.. note::
+  Each concept that concretizes a dataset definition (like different tables or
+  data sources) shall contain additional metadata which describes this concept
+  as the whole. For example, :ref:`data-source` and :ref:`table` concepts have
+  different sets of metadata since they represent different kinds of a
+  :term:`dataset`.
+
+.. _table-builder:
+
+Table builder
+-------------
+
+A table :term:`builder` is a concept that associated with particular
+:ref:`table` type and is used on data-preparation and data-processing stages
+for:
+
+- Iterative (in general case) construction of a :ref:`table` from
+  another :ref:`tables <table>` or different in-memory :ref:`dataset`
+  representations
+
+- Construction of a :ref:`table` from different entities that hold pieces of the
```

Comment:
`entities that hold pieces of the dataset` seems unclear
other entities that contain information about the data? not sure it's better